src/main/kotlin/de/cyface/collector/storage/cloud/MongoDatabase.kt

```diff
@@ -52,23 +53,21 @@ class MongoDatabase(private val mongoClient: MongoClient, private val collection
     }
 
     /**
-     * Formats the provided [UploadMetaData] into a [JsonObject] which can be stored in the database.
-     *
-     * Currently, we maintain a schema consistent with the one used in GridFS storage to leverage existing toolchains.
-     * Note that `chunkSize` is omitted as blobs are stored as single files in the object store.
+     * Transforms [UploadMetaData] into a the database format.
      *
      * @param metaData The [UploadMetaData] to format.
-     * @return The [JsonObject] representation of the [UploadMetaData].
+     * @return The GeoJSON representation of the [UploadMetaData].
      */
     fun databaseFormat(metaData: UploadMetaData): JsonObject {
-        val isoDate = DateTimeFormatter.ISO_INSTANT.format(Instant.now())
-        val storageMeta = metaData.uploadable.toJson()
+        val json = metaData.uploadable.toGeoJson()
+
+        json.getJsonObject("properties")
             .put(USER_ID_DATABASE_FIELD, metaData.user.id.toString())
-        return JsonObject()
             .put("filename", metaData.uploadIdentifier.toString()) // The id/name of the file in the object storage
-            .put("length", metaData.contentRange.totalBytes) // blob size in Bytes (Long)
-            .put("uploadDate", JsonObject().put("\$date", isoDate))
-            .put("metadata", storageMeta)
+            .put("uploadLength", metaData.contentRange.totalBytes) // measurement blob size in Bytes (Long)
+            .put("uploadDate", JsonObject().put("\$date", DateTimeFormatter.ISO_INSTANT.format(Instant.now())))
```

Comment:
This information is only available at this point, so I add this information to the metadata before storing it in the database.

We need:
- userId (obviously)
- filename: to identify the object storage blob for the metadata
- uploadLength: This was called `length` before but would now collide with the old `metadata.length` (distance of the track). But I think we should still store the blob size in mongo, so that we do not have to query the object store just to show the measurement size e.g. each time the web app shows details about a measurement. Removing this would probably break other features down the stream (backend/web app).
- uploadDate: This is also shown in the web app, so I keep this info.